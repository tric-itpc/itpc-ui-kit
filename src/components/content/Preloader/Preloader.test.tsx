import React from "react"

import { render, screen } from "@testing-library/react"

import { Preloader } from "./index"

describe("Компонент Preloader", () => {
  test("Рендер с обязательными пропсами", () => {
    render(<Preloader />)
    const preloader = screen.getByRole("preloader")
    expect(preloader).toBeInTheDocument()
    expect(preloader).toHaveClass("itpc-preloader")
  })

  test("Применение дополнительных классов", () => {
    render(<Preloader className="custom-class" />)
    const preloader = screen.getByRole("preloader")
    expect(preloader).toHaveClass("custom-class")
  })

  test("Применение дополнительных атрибутов", () => {
    render(<Preloader aria-label="loading" className="custom-class" />)
    const preloader = screen.getByLabelText("loading")
    expect(preloader).toBeInTheDocument()
    expect(preloader).toHaveClass("itpc-preloader", "custom-class")
  })

  test("Снапшот с обязательными пропсами", () => {
    const { asFragment } = render(<Preloader />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("Снапшот с дополнительным классом", () => {
    const { asFragment } = render(<Preloader className="custom-class" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("Снапшот с дополнительными атрибутами", () => {
    const { asFragment } = render(
      <Preloader aria-label="loading" className="custom-class" />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
