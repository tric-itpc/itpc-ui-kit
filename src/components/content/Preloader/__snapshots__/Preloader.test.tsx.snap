// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Preloader Снапшот с дополнительным классом 1`] = `
<DocumentFragment>
  <span
    class="itpc-preloader custom-class"
    role="preloader"
  />
</DocumentFragment>
`;

exports[`Компонент Preloader Снапшот с дополнительными атрибутами 1`] = `
<DocumentFragment>
  <span
    aria-label="loading"
    class="itpc-preloader custom-class"
    role="preloader"
  />
</DocumentFragment>
`;

exports[`Компонент Preloader Снапшот с обязательными пропсами 1`] = `
<DocumentFragment>
  <span
    class="itpc-preloader"
    role="preloader"
  />
</DocumentFragment>
`;
